deployments/cloudfoundry/buildpack/README.md

```diff
@@ -0,0 +1,155 @@
+- `OTELCOLLECTOR_DOWNLOAD_URL` - URL to download the OpenTelemetry Collector from.
+
+- `RLP_GATEWAY_SHARD_ID` - Metrics are load balanced between receivers that use the same shard ID.
+   Only use if multiple receivers must receive all metrics instead of
+   balancing metrics between them. Default: `otelcol`
+- `RLP_GATEWAY_TLS_INSECURE` - Whether to skip TLS verify for the RLP gateway endpoint. Default: `true`
+- `UAA_TLS_INSECURE` - Whether to skip TLS verify for the UAA endpoint. Default: `true`
+- `SIGNALFX_INGEST_URL` - The ingest base URL for SignalFx. If specified, SIGNALFX_API_URL is also required.
+   This option takes precedence over SIGNALFX_REALM.
+- `SIGNALFX_API_URL` - The API server base URL for SignalFx. If specified, SIGNALFX_INGEST_URL is also required.
+   This option take precedence over SIGNALFX_REALM.
+- `SIGNALFX_REALM` - The SignalFx realm in which your organization resides. Used to derive SIGNALFX_INGEST_URL and
+   SIGNALFX_API_URL.
+
+## Sidecar Configuration
+
+The recommended method for running the Collector is to run it as a sidecar using
+the Cloud Foundry [sidecar
+functionality](https://docs.cloudfoundry.org/devguide/sidecars.html).
+Additional information can be found [in the v3 API
+docs](http://v3-apidocs.cloudfoundry.org/version/release-candidate/#sidecars).
+
+Here is an example application `manifest.yml` file that would run the Collector as
+a sidecar:
+
+```yaml
+---
+applications:
+  - name: test-app
+    buildpacks:
+      - otel_collector_buildpack
+      - go_buildpack
+    instances: 1
+    memory: 256M
+    random-route: true
+    env:
+      RLP_GATEWAY_ENDPOINT: "https://log-stream.sys.<TAS environment name>.cf-app.com"
+      UAA_ENDPOINT: "https://uaa.sys.<TAS environment name>.cf-app.com"
+      UAA_USERNAME: "..."
+      UAA_PASSWORD: "..."
+      SIGNALFX_ACCESS_TOKEN: "..."
+      SIGNALFX_REALM: "..."
+    sidecars:
+      - name: otel-collector
+        process_types:
+          - web
+        command: "$HOME/../deps/otelcontribcol_${OS:-linux_amd64}-v${OTEL_VERSION:-0.47.0} --config=$HOME/../deps/${OTEL_CONFIG:-otelconfig.yaml}"
+        memory: 100MB
+```
+If using a `manifest.yaml` file, you may push your app simply with the following command:
+```sh
+# If you are using cf CLI v7
+$ cf push
+
+# If you are using cf CLI v6
+$ cf v3-push <app-name>
+```
+This will deploy the app with the proper buildpacks, and the OpenTelemetry Collector running in the sidecar configuration.
+
+## Troubleshooting
+
+* If the app is running but the OpenTelemetry Collector is not, it may be that the sidecar configuration is not
+being picked up properly from the manifest file. Try running the following commands:
+
+```sh
+# This will apply the manifest file to an existing app
+$ cf v3-apply-manifest -f manifest.yml
+# This will re-load the app with the sidecar configuration included
+$ cf push
+```
+
+* Another possibility is the sidecar was not allocated memory. The `memory` option
```

Comment:
possible for an example?

Reply:
I can include an example yaml block for the sidecar without the memory specified explicitly, but that's all I can really think of as an example here.